modules/javafx.graphics/src/main/java/com/sun/javafx/iio/javax/XImageLoader.java

```diff
@@ -0,0 +1,283 @@
+            null, null);
+
+        return switch (image.getType()) {
+            case TYPE_BYTE_GRAY -> new ImageFrame(ImageStorage.ImageType.GRAY,
+                    getByteBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth(),
+                    null, pixelScale, metadata);
+
+            case TYPE_3BYTE_BGR -> new ImageFrame(ImageStorage.ImageType.BGR,
+                    getByteBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 3,
+                    null, pixelScale, metadata);
+
+            case TYPE_4BYTE_ABGR -> new ImageFrame(ImageStorage.ImageType.ABGR,
+                    getByteBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            case TYPE_4BYTE_ABGR_PRE -> new ImageFrame(ImageStorage.ImageType.ABGR_PRE,
+                    getByteBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            case TYPE_INT_RGB -> new ImageFrame(ImageStorage.ImageType.INT_RGB,
+                    getIntBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            case TYPE_INT_BGR -> new ImageFrame(ImageStorage.ImageType.INT_BGR,
+                    getIntBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            case TYPE_INT_ARGB -> new ImageFrame(ImageStorage.ImageType.INT_ARGB,
+                    getIntBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            case TYPE_INT_ARGB_PRE -> new ImageFrame(ImageStorage.ImageType.INT_ARGB_PRE,
+                    getIntBuffer(image.getData().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * 4,
+                    null, pixelScale, metadata);
+
+            default ->
+                throw new ImageStorageException("Unsupported image type: " + switch (image.getType()) {
+                    case TYPE_CUSTOM -> "TYPE_CUSTOM";
+                    case TYPE_USHORT_565_RGB -> "TYPE_USHORT_565_RGB";
+                    case TYPE_USHORT_555_RGB -> "TYPE_USHORT_555_RGB";
+                    case TYPE_USHORT_GRAY -> "TYPE_USHORT_GRAY";
+                    case TYPE_BYTE_BINARY -> "TYPE_BYTE_BINARY";
+                    case TYPE_BYTE_INDEXED -> "TYPE_BYTE_INDEXED";
+                    default -> Integer.toString(image.getType());
+                });
```

Comment:
Yes, but I don't know which modern image format that people would want to load is using indexed colors. If it turns out that there is a pressing need, this might be better split off into another work package.

SVG produces `TYPE_INT_ARGB_PRE` in my tests.

Reply:
TYPE_BYTE_INDEXED is used in common TIFF ImageIO readers for monochrome compressions.